TestsFoundation/TestsFoundation/Extensions/XCUIElementExtensions.swift

```diff
@@ -19,50 +19,241 @@
 import Foundation
 import XCTest
 
+public var app: XCUIApplication { XCUIApplication() }
+
 public extension XCUIElement {
-    func find(label: String, type: XCUIElement.ElementType = .any) -> Element {
-        return descendants(matching: type).matching(label: label).firstElement
+    static let defaultTimeout: TimeInterval = 15
+    static let defaultGracePeriod: UInt32 = 1
+    var isVisible: Bool { exists }
+    var isVanished: Bool { !(exists && isHittable) }
+    func tacticalSleep(ms: UInt32 = 500000) { usleep(ms) }
+
+    enum ElementCondition {
+        case visible
+        case vanish
+        case value
+        case label
+        case enabled
+        case selected
+        case unselected
+        case hittable
+        case labelContaining
+        case labelHasPrefix
+    }
+
+    enum ElementAction {
+        case swipeUp
+        case swipeDown
+        case swipeRight
+        case swipeLeft
+        case tap
+        case showKeyboard
+        case hideKeyboard
+        case pullToRefresh
+    }
+
+    func hasValue(value: String, strict: Bool = true) -> Bool {
+        let elementValue = self.value as? String ?? ""
+        return strict ? elementValue == value : elementValue.contains(value)
+    }
+
+    func hasLabel(label: String, strict: Bool = true) -> Bool {
+        let elementLabel = self.label
+        return strict ? elementLabel == label : elementLabel.contains(label)
+    }
+
+    @discardableResult
+    func hit() -> XCUIElement {
+        waitUntil(condition: .visible)
+        if !isHittable { actionUntilElementCondition(action: .swipeUp, condition: .hittable, timeout: 5) }
+        tap()
+        return self
+    }
+
+    @discardableResult
+    func waitUntil(condition: ElementCondition,
```

Comment:
We could turn this `condition` parameter to anonym `func waitUntil(_ condition: ElementCondition` so calling it would be shorter and I think easier to read: `waitUntil(.visible)`.

Reply:
Done.